test/useObservableProps.test.tsx

```diff
@@ -1,9 +1,7 @@
 import * as mobx from "mobx"
 import * as React from "react"
+import { render } from "react-testing-library"
```

Comment:
I haven't studied this file closely since it will be most likely dropped, but it's super flaky. Each test should be isolated and not depend on a previous one :) That's why there is `afterEach(cleanup)` in other tests.

Reply:
The thing is that they actually depend since each "test" is based on rerenders of the component rather than renders from 0

I'll set cleanups on the big chunks though

Reply:
Well, still it shouldn't depend. If it's really necessary then it's better to have one bigger test than add then depend on the previous one. Have a look at other tests. I usually have a setup function that's executed at beginning of each test and then does what's necessary for each test.

Reply:
Nonetheless, the `useObservableProps` should be removed altogether, so no point in updating tests for it unless you want to keep it and publish in a separate repo/package.

Reply:
That's ok :) I appreciate the reviews nevertheless

I'm starting to think that the most useful thing about useObservableProps is the "object assign" between an observer object and plain properties with the reconciliation algorithm, so maybe its place might be into mobx-utils?

For example, it can be used when you make a fetch to some API and want to update your observable representation of that object without triggering reactions for changes that didn't really change

Reply:
I just accepted into this package in other comment and you change your mind in this one 😄🥇 

Ok, let's move it into a separate package since the use cases for it are rather sparse imo.

Reply:
removed useObservableProps